File: 7.reverse-integer.py
#
# @lc app=leetcode id=7 lang=python3
#
# [7] Reverse Integer
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        sym = 1

        if x < 0:
            x = -x
            sym = -1

        while x:
            res *= 10
            res += x%10
            x //= 10

        res *= sym

        if res > pow(2,31) - 1 or res < -pow(2,31): return 0
        
        return res
        
# @lc code=end

File: 7.reverse-integer.cpp
/*
 * @lc app=leetcode id=7 lang=cpp
 *
 * [7] Reverse Integer
 */

// @lc code=start
class Solution {
public:
    int reverse(int x) {
        long int res = 0;
        long int sign = x >= 0 ? 1 : -1;
        long int xx = sign*x;
        while (xx)
        {
            long int temp = xx%10;
            xx /= 10;
            res *= 10;
            res += temp;
        }
        res *= sign;
        if (res > INT_MAX || res < INT_MIN) return 0;
        return res;
    }
};
// @lc code=end

